src/engine/Server.cpp

```diff
@@ -983,12 +992,14 @@ Awaitable<void> Server::processQueryOrUpdate(
   // access to the runtimeInformation in the case of an error.
   std::optional<PlannedQuery> plannedQuery;
   try {
-    if constexpr (type == OperationType::Query) {
-      co_await processQuery(params, queryOrUpdate, requestTimer, request, send,
+    if constexpr (std::is_same_v<Operation, Query>) {
+      co_await processQuery(params, operation, requestTimer, request, send,
                             timeLimit);
-    } else {
-      co_await processUpdate(params, queryOrUpdate, requestTimer, request, send,
+    } else if constexpr (std::is_same_v<Operation, Update>) {
+      co_await processUpdate(params, operation, requestTimer, request, send,
                              timeLimit);
+    } else {
+      AD_FAIL();
```

Comment:
Can this statically happen?
Otherwise you can replace the `else if constexpr` by `else static_assert`

Reply:
What do you mean by doing this statically? The comment is now applied and also a `static_assert(std::is_same_v<Operation, Query> || std::is_same_v<Operation, Update>)` at the beginning (which my IDE marked at some time). What would be a good style for doing this (the `static_assert` in the else is redundant now)?

Reply:
I don't mind the redundant static assert in the beginning.
The one in the beginning gives nice documentation + error messages,
And in the if-else it is useful to prevent bugs etc. So I really like the way it is now.